python/cudf/cudf/core/window/rolling.py

```diff
@@ -557,5 +557,6 @@ def _apply_agg(self, agg_name):
             )
         )
 
-        result = super()._apply_agg(agg_name).set_index(index)
+        result = super()._apply_agg(agg_name)
+        result.index = index
```

Comment:
Is it the case that `super()._apply_agg(agg_name)` will always produce a `Series` here? Because if it can produce a `DataFrame`, this change seems bad (`DataFrame.set_index` does significantly more than just assigning the `.index` attribute)

Reply:
For objects of `Index`/`MultiIndex` type, either way leads to the same results:
```python
In [9]: df = cudf.DataFrame({'a':[10, 11, 12, 13]}, index=['a', 'b', 'c', 'd'])

In [10]: df
Out[10]: 
    a
a  10
b  11
c  12
d  13

In [11]: df.set_index(cudf.Index([100, 200, 300, 400]))
Out[11]: 
      a
100  10
200  11
300  12
400  13

In [12]: df = cudf.DataFrame({'a':[10, 11, 12, 13]}, index=['a', 'b', 'c', 'd'])

In [13]: df.index = cudf.Index([100, 200, 300, 400])

In [14]: df
Out[14]: 
      a
100  10
200  11
300  12
400  13
```